src/plugins/code-golf/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { Component, jsx } from "#DCGView";
+import { Inserter, PluginController } from "../PluginController";
+import "./index.less";
+import { format } from "localization/i18n-core";
+import { IfElse } from "src/components";
+import { ExpressionModel } from "src/globals";
+
+export class ExpressionItemCostPanel extends Component<{
+  model: () => ExpressionModel;
+  el: () => HTMLDivElement;
+}> {
+  rootblock: Element | null | undefined = null;
+
+  template() {
+    const chars = () => this.props.model().latex?.length ?? 0;
+
+    return IfElse(() => chars() > 0, {
+      true: () => (
+        <div class="dsm-code-golf-char-count-container">
+          <div class="dsm-code-golf-char-count">
+            <div>
+              {() => {
+                const tempRootblock = this.props
+                  .model()
+                  .dcgView?._element._domNode?.querySelector(
+                    ".dcg-main .dcg-mq-root-block"
+                  );
+
+                if (tempRootblock) this.rootblock = tempRootblock;
+
+                if (!this.rootblock) return "0px";
+
+                if (!this.rootblock.lastChild || !this.rootblock.firstChild)
+                  return "0px";
```

Comment:
Are these supposed to be unreachable? If there's a chance of being reachable, might want to `format("code-golf-width-in-pixels", {pixels: "0"})` in a similar manner as defaulting to zero symbol count below.

Reply:
They were reachable initially and yeah I agree I should `format` them. That being said, I *suspect* that they're unreachable now.

Reply:
Those were supposed to be reachable and yeah I should really `format` them.

However they might be unreachable due to other changes I made.

Reply:
oh wait i just realized I answered this in a different tab already whoopsie lol

Reply:
Nice, I like the new way where it's a separate function for computing the width.